File: .repo/system-design-mastery-module-2-database-fundamentals/0-sql-vs-nosql/1-java/src/main/java/academy/starci/sqlvsnosql/nosql/NosqlController.java
package academy.starci.sqlvsnosql.nosql;

import academy.starci.sqlvsnosql.dto.CreateProductDto;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/nosql")
public class NosqlController {

    private final NosqlService nosqlService;

    public NosqlController(NosqlService nosqlService) {
        this.nosqlService = nosqlService;
    }

    @PostMapping("/products")
    public ResponseEntity<ProductDocument> create(@RequestBody CreateProductDto dto) {
        ProductDocument created = nosqlService.create(dto);
        return ResponseEntity.status(HttpStatus.CREATED).body(created);
    }

    @GetMapping("/products")
    public List<ProductDocument> findAll() {
        return nosqlService.findAll();
    }

    @GetMapping("/products/category/{cat}")
    public List<ProductDocument> findByCategory(@PathVariable("cat") String category) {
        return nosqlService.findByCategory(category);
    }
}

File: .repo/system-design-mastery-module-2-database-fundamentals/0-sql-vs-nosql/2-csharp/Controllers/NosqlController.cs
using Microsoft.AspNetCore.Mvc;
using SqlVsNosql.Models;
using SqlVsNosql.Services;
using System.Threading.Tasks;

namespace SqlVsNosql.Controllers
{
    [ApiController]
    [Route("api/nosql")]
    public class NosqlController : ControllerBase
    {
        private readonly NosqlService _nosqlService;

        public NosqlController(NosqlService nosqlService)
        {
            _nosqlService = nosqlService;
        }

        [HttpPost("products")]
        public async Task<IActionResult> Create([FromBody] CreateProductDto dto)
        {
            var created = await _nosqlService.CreateAsync(dto);
            return Created($"/api/nosql/products/{created.Id}", created);
        }

        [HttpGet("products")]
        public async Task<IActionResult> FindAll()
        {
            var products = await _nosqlService.FindAllAsync();
            return Ok(products);
        }

        [HttpGet("products/category/{cat}")]
        public async Task<IActionResult> FindByCategory([FromRoute(Name = "cat")] string category)
        {
            var products = await _nosqlService.FindByCategoryAsync(category);
            return Ok(products);
        }
    }
}
